src/nspawn/nspawn.c

```diff
@@ -1018,14 +1011,17 @@ static int parse_argv(int argc, char *argv[]) {
                         assert_not_reached("Unhandled option");
                 }
 
+        if (getenv_bool("SYSTEMD_NSPAWN_SHARE_SYSTEM") > 0)
+                arg_share_system = true;
```

Comment:
I think almost nobody is using it right now, and the ones who want to use it really want something different actually. For example, people asked in order to be able to map kubernetes container model, they want to select UTC, IPC, PID namespacing individually, but the current option does that. I am willing to add support for this kubernetes logic, but i really wouldn't do that as primary option, but only via env vars, since it's a really low-level option and it blurs the lines too much between containers and services I think...

Currently, there are already quite a few places where the sandboxing of services and of nspawn can do similar things. I'd really like to make sure we have at least a vague idea where nspawn territory begins, and where service management territory ends. My suggestion would be to bind this around PID namespacing really: if PID namespacing is used, then it is nspawn territory, and if it is not used, then it is systemd service management territory. If this is an acceptable line, then I think --share-system really should go, as it contradicts this line prominently.

So, all in all, I really think this option should go away: to my knowledge it isn't used, and it shouldn't be used. It is now useful for those who actually want a feature like this, and it contradicts or message I think.

In order to make the transition easy for those who use it currently (if there really are people like that), I kept the mechanism in place, reachable via an env var instead of a primary switch.

I hope this makes sense